chore: Adding example for .NET minimal APIs

aeliaNR opened the pull request:
*I am a New Relic employee so please feel free to ping me via Slack with question.*

This PR adds an example .NET application which implements OpenTelemetry via the [Minimal API](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis?view=aspnetcore-6.0) paradigm.

jack-berg commented on dotnet/minimal-api/.gitignore:
```diff
@@ -0,0 +1,353 @@
+# Do not track changes to env file after placeholder
```
No need for a `.gitignore` in a subdirectory

jack-berg commented on dotnet/minimal-api/Dockerfile:
```diff
@@ -0,0 +1,20 @@
+WORKDIR /app
+COPY --from=build-env /app/out .
+ENV OTEL_EXPORTER_OTLP_ENDPOINT = ${OTEL_EXPORTER_OTLP_ENDPOINT}
+ENV OTEL_EXPORTER_OTLP_HEADERS = ${OTEL_EXPORTER_OTLP_HEADERS}
```
Let's remove these `ENV` lines and instead just read the uesr's system environment variables in in `docker-compose.yml`

aeliaNR replied:
I hesitate on this one in part because by leaving these lines we reinforce our need for the environment variables to be passed without causing harm/confusion.

jack-berg commented on dotnet/minimal-api/Dockerfile:
```diff
@@ -0,0 +1,20 @@
+ARG OTEL_EXPORTER_OTLP_ENDPOINT
+ARG OTEL_EXPORTER_OTLP_HEADERS
```
```suggestion
```

jack-berg commented on dotnet/minimal-api/README.md:
```diff
@@ -0,0 +1,25 @@
+  - OTEL_EXPORTER_OTLP_ENDPOINT
+    - Set this to `https://otlp.nr-data.net:4317`
+  - OTEL_EXPORTER_OTLP_HEADERS
+    - Set this to `api-key=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX` replacing `XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX` with a valid New Relic Ingest License key found [here](https://one.newrelic.com/admin-portal/api-keys/home).
```
#nit: Let's copy the environment variable specification from [here](https://github.com/newrelic/newrelic-opentelemetry-examples/tree/main/dotnet/aspnetcore#run-the-application).

jack-berg commented on dotnet/minimal-api/README.md:
```diff
@@ -0,0 +1,25 @@
+## Running locally
+
+- If you set the environment variables on your local environment, you can run this program via `dotnet run` from a terminal instances in this directory.
+- If you set the environment variables in the [.env](.env) file then you can run this program via `docker compose --env-file .env up`
```
Let's remove the requirement on `.env`. See my recommendation in `docker-compose.yml`.

jack-berg commented on dotnet/minimal-api/docker-compose.yml:
```diff
@@ -0,0 +1,9 @@
+    ports:
+    - "8080:80"
+    build: .
+    env_file:
+    - .env
```
```suggestion
    environment:
      OTEL_EXPORTER_OTLP_ENDPOINT: "${OTEL_EXPORTER_OTLP_ENDPOINT}"
      OTEL_EXPORTER_OTLP_HEADERS: "${OTEL_EXPORTER_OTLP_HEADERS}"
```

aeliaNR replied:
Resolved in [321ae45](https://github.com/newrelic/newrelic-opentelemetry-examples/pull/99/commits/321ae452480435cbf14e03dcaea2ad8f59cdd8ce)

jack-berg reviewed (commented):
Thanks for the submission! Left a few minor comments.

TylerHelmuth commented:
Merged this branch with origin/main to get the latest workflows

aeliaNR commented:
@jack-berg - I believe I have addressed all of your feedback via [321ae45](https://github.com/newrelic/newrelic-opentelemetry-examples/pull/99/commits/321ae452480435cbf14e03dcaea2ad8f59cdd8ce). Please review and advise. Thank you so much for your attention to and effort in reviewing this PR.
